File: backend/src/main/java/dev/fogmap/social/SocialDtos.java
package dev.fogmap.social;

import jakarta.validation.constraints.NotBlank;

public final class SocialDtos {

    private SocialDtos() {
    }

    public record FriendRequestBody(@NotBlank String username) {
    }

    public record FriendDto(long userId, String username, long areaM2) {
    }

    public record PendingRequestDto(long userId, String username) {
    }

    public record LeaderboardEntryDto(int rank, long userId, String username, long areaM2) {
    }

    public record AchievementDto(String code, String title, long unlockedAt) {
    }

    public record UserStatsDto(long userId, String username, long areaM2) {
    }
}

File: android/data/src/main/kotlin/dev/fogmap/data/api/SocialDtos.kt
package dev.fogmap.data.api

import kotlinx.serialization.Serializable

@Serializable
data class Friend(val userId: Long, val username: String, val areaM2: Long)

@Serializable
data class PendingRequest(val userId: Long, val username: String)

@Serializable
data class LeaderboardEntry(
    val rank: Int,
    val userId: Long,
    val username: String,
    val areaM2: Long,
)

@Serializable
data class Achievement(val code: String, val title: String, val unlockedAt: Long)

@Serializable
internal data class FriendRequestBody(val username: String)
